webapp/channels/src/components/post_view/post_flag_icon/post_flag_icon.tsx

```diff
@@ -19,115 +19,98 @@ export type Actions = {
     unflagPost: typeof unflagPost;
 }
 
-interface Props {
+type Props = {
     location?: keyof typeof Locations;
     postId: string;
     isFlagged: boolean;
     actions: Actions;
 }
 
-interface State {
-    a11yActive: boolean;
-}
-
-export default class PostFlagIcon extends React.PureComponent<Props, State> {
-    static defaultProps = {
-        location: Locations.CENTER,
-    };
-
-    private buttonRef: React.RefObject<HTMLButtonElement>;
-
-    constructor(props: Props) {
-        super(props);
-
-        this.buttonRef = React.createRef();
-
-        this.state = {
-            a11yActive: false,
-        };
-    }
-
-    componentDidMount() {
-        if (this.buttonRef.current) {
-            this.buttonRef.current.addEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
-            this.buttonRef.current.addEventListener(A11yCustomEventTypes.DEACTIVATE, this.handleA11yDeactivateEvent);
-        }
-    }
-    componentWillUnmount() {
-        if (this.buttonRef.current) {
-            this.buttonRef.current.removeEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
-            this.buttonRef.current.removeEventListener(A11yCustomEventTypes.DEACTIVATE, this.handleA11yDeactivateEvent);
-        }
-    }
-
-    componentDidUpdate() {
-        if (this.state.a11yActive && this.buttonRef.current) {
-            this.buttonRef.current.dispatchEvent(new Event(A11yCustomEventTypes.UPDATE));
-        }
-    }
-
-    handlePress = (e: React.MouseEvent) => {
+const PostFlagIcon = ({
+    actions: {
+        flagPost,
+        unflagPost,
+    },
+    isFlagged,
+    postId,
+    location = Locations.CENTER,
+}: Props) => {
+    const buttonRef = useRef<HTMLButtonElement>(null);
+    const [a11yActive, setA11yActive] = useState(false);
+
+    const handlePress = useCallback((e: React.MouseEvent) => {
```

Comment:
`useCallback` to avoid passing a new function to children objects on every render.